tests/probes_test.go

```diff
@@ -147,6 +126,38 @@ var _ = Describe("[ref_id:1182]Probes", func() {
 	})
 })
 
+func createReadyCirrosVMIWithReadinessProbe(virtClient kubecli.KubevirtClient, probe *v12.Probe) *v12.VirtualMachineInstance {
+	dummyUserData := "#!/bin/bash\necho 'hello'\n"
+	vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(
+		cd.ContainerDiskFor(cd.ContainerDiskCirros), dummyUserData)
+	vmi.Spec.ReadinessProbe = probe
+
+	return createAndBlockUntilVMIHasStarted(virtClient, vmi)
+}
+
+func createReadyCirrosVMIWithLivenessProbe(virtClient kubecli.KubevirtClient, probe *v12.Probe) *v12.VirtualMachineInstance {
+	dummyUserData := "#!/bin/bash\necho 'hello'\n"
+	vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(
+		cd.ContainerDiskFor(cd.ContainerDiskCirros), dummyUserData)
+	vmi.Spec.LivenessProbe = probe
+
+	return createAndBlockUntilVMIHasStarted(virtClient, vmi)
+}
+
+func createAndBlockUntilVMIHasStarted(virtClient kubecli.KubevirtClient, vmi *v12.VirtualMachineInstance) *v12.VirtualMachineInstance {
+	_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+	Expect(err).ToNot(HaveOccurred())
+
+	// It may come to modify retries on the VMI because of the kubelet updating the pod, which can trigger controllers more often
+	tests.WaitForSuccessfulVMIStartIgnoreWarnings(vmi)
+
+	// assure the VMI has a UUID available
+	startedVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &v13.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(startedVMI.UID).NotTo(BeEmpty())
```

Comment:
Why does this extra check needed?

Reply:
Weeeeeell ... It isn't. 

I just need the VMI to have a UUID available in it. Simply 'GETting' it from the cluster would suffice.